### ISSUE TITLE
gethostname linkage fails, even though present in wasi-libc

### ISSUE THREAD

**Issue by marmistrz** [WebAssembly/wasi-libc#196]
I tried to compile the following code using wasi-sdk
```
clang hostname.c --target=wasm32-wasi --sysroot=/opt/wasi-sdk/wasi-sysroot
```
```
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

int main(void) {

    char hostname[1024];
    gethostname(hostname, 1024);

    puts(hostname);

    return EXIT_SUCCESS;
}
```
This fails during the linking phase:
```
wasm-ld: error: /tmp/hostname-bd4ad1.o: undefined symbol: gethostname
clang-10: error: linker command failed with exit code 1 (use -v to see invocation)
```
even though `gethostname` is implemented by `wasi-libc`.

**Comment by sunfishcode** [comment]
While the musl netdb.h and gethostbyname implemenation are present in the tree, they are disabled in the WASI libc build. WASI does not yet have API support needed to implement `gethostbyname`.

**Comment by marmistrz** [comment]
That's what I expected because I wasn't aware of such API either. In such case, we should mention it in gethostbyname.c et al.

**Comment by sbc100** [comment]
Isn't the solution to remove the functions/headers so this becomes a compile error rather than a link error?

**Comment by sunfishcode** [comment]
Agreed, we should do that. This would also be a good function to add as an optional "emulated" function, and have it return some fixed string like "wasihost" or something for compatibility with code that doesn't actually need it to mean anything.
